trigger/kafkasub/trigger.go

```diff
@@ -119,49 +121,53 @@ func run(t *KafkaSubTrigger) {
 		panic(fmt.Errorf("Failed to create Kafka consumer for reason [%s]", error))
 	}
 	t.kafkaConsumer = &kafkaConsumer
-	thing := make(map[int32]sarama.PartitionConsumer)
-	t.partitionConsumers = &thing
-	flogoLogger.Debugf("Subscribing to topic [%s]", t.kafkaParms.topic)
-	validPartitions, error := kafkaConsumer.Partitions(t.kafkaParms.topic)
-	if error != nil {
-		panic(fmt.Errorf("Failed to get valid partitions from Kafka Consumer for reason [%s].  Aborting subscriber", error))
-	}
-
-	flogoLogger.Debugf("Valid partitions for topic [%s] detected as: [%v]", t.kafkaParms.topic, validPartitions)
-	if t.kafkaParms.partitions == nil { //subscribe to all valid partitions
-		t.kafkaParms.partitions = validPartitions
-		for _, part := range validPartitions {
-			startConsumer(t, part)
+	consumers := make(map[string]sarama.PartitionConsumer)
+	t.partitionConsumers = &consumers
+	for id, handler := range t.kafkaParms.handlers {
+		validPartitions, error := kafkaConsumer.Partitions(handler.topic)
+		if error != nil {
+			panic(fmt.Errorf("Failed to get valid partitions for topic [%s] for reason [%s].  Aborting subscriber",
+				error))
 		}
-	} else { //subscribe to a subset of valid partitions
-		configPartitions := t.kafkaParms.partitions
-		for _, confPart := range configPartitions {
-			for _, valPart := range validPartitions {
-				if confPart == valPart {
-					startConsumer(t, confPart)
-					break
+		flogoLogger.Debugf("Subscribing to topic [%s]", handler.topic)
+
+		flogoLogger.Debugf("Valid partitions for topic [%s] detected as: [%v]", handler.topic, validPartitions)
+		if handler.partitions == nil { //subscribe to all valid partitions
+			handler.partitions = validPartitions
+			for _, part := range validPartitions {
+				startConsumer(t, part, id)
+			}
+		} else { //subscribe to a subset of valid partitions
+			configPartitions := handler.partitions
+			for _, confPart := range configPartitions {
+				for _, valPart := range validPartitions {
+					if confPart == valPart {
+						startConsumer(t, confPart, id)
+						break
+					}
+					flogoLogger.Errorf("Configured partition [%d] on topic [%s] does not exist and will not be subscribed", confPart, handler.topic)
 				}
 			}
-			flogoLogger.Errorf("Configured partition [%d] on topic [%s] does not exist", confPart, t.kafkaParms.topic)
 		}
-	}
-	if len(*t.partitionConsumers) < 1 {
-		panic(fmt.Errorf("Kafka consumer is not configured for any valid partitions"))
-	} else {
-		flogoLogger.Debugf("Kafka consumers for topic [%s] started", t.kafkaParms.topic)
+		if len(*t.partitionConsumers) < 1 {
+			panic(fmt.Errorf("Kafka consumer is not configured for any valid partitions"))
```

Comment:
Start() in a trigger should not panic, just return an error and the engine will handle it.